modules/c++/mt/include/mt/WorkSharingBalancedRunnable1D.h

```diff
@@ -271,8 +271,8 @@ void runWorkSharingBalanced1D(size_t numElements,
                       mem::SharedPtr<sys::AtomicCounter>(
                               new sys::AtomicCounter(startElement)));
 
-              threadPoolEndElements[threadNum++] =
```

Comment:
It looks like `threadNum` needs to be incremented at the top of this loop in `planner.getThreadInfo(..`

Reply:
Oh jeez - thanks for catching this.  Fixed.